File: tanxin/3202找出有效子序列的最大长度II.py
from typing import List
class Solution:
    def maximumLength(self, nums: List[int], k: int) -> int:
        dp=[[0]*k for _ in range(0,k)]
        for x in nums:
            x%=k
            for y in range(0,k):
                dp[y][x]=dp[x][y]+1
        return max(map(max,dp))

File: tanxin/3202找出有效子序列的最大长度II.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int maximumLength(vector<int>& nums, int k) {
        vector<vector<int>>dp(k,vector<int>(k,0));
        for(auto x:nums){
            x%=k;
            for(int y=0;y<k;y++){
                dp[y][x]=dp[x][y]+1;
            }
        }
        int maximum=0;
        for(auto i:dp){
            for(auto j:i){
                maximum=max(j,maximum);
            }
        }
        return maximum;
    }
};